packages/wallet-sdk/src/CoinbaseWalletProvider.ts

```diff
@@ -47,8 +48,24 @@ export class CoinbaseWalletProvider extends ProviderEventEmitter implements Prov
       if (!this.signer) {
         switch (args.method) {
           case 'eth_requestAccounts': {
-            const signerType = await this.requestSignerSelection(args);
+            let signerType: SignerType;
+            const subAccountsConfig = store.subAccountsConfig.get();
+            if (subAccountsConfig?.enableAutoSubAccounts) {
+              signerType = 'scw';
+            } else {
+              signerType = await this.requestSignerSelection(args);
+            }
             const signer = this.initSigner(signerType);
+
+            if (signerType === 'scw' && subAccountsConfig?.enableAutoSubAccounts) {
+              await signer.handshake({ method: 'handshake' });
+
+              const result = await signer.request(args);
+              this.signer = signer;
+
+              return result as T;
+            }
+
```

Comment:
L:63 seems to be still passing requestAccounts over to FE? believe we should replace it with `wallet_connect` here?

Reply:
ahhh makes sense

Reply:
i wonder if we should move the conversion logic to `SCWSigner.handshake` for when arg.method is `eth_requestAccounts`? and have the `SCWSigner.request(eth_requestAccounts)` to be just fetching from cache

Reply:
good idea, i like that better

Reply:
i cleaned this up and refactored the handler in `SCWSigner` to preserve the caching behavior with auto sub accounts